torch/_functorch/partitioners.py

```diff
@@ -2515,11 +2515,38 @@ def has_same_nodes(joint_graph):
     ):
         with no_dispatch(), unset_fake_temporarily():
             objects = [[x.name for x in saved_values]]
-            # TODO: maybe use a different process group for this
-            torch.distributed.broadcast_object_list(objects, src=0)
-            saved_values_names = objects[0]
+            saved_ops_names_all_ranks: list[list[str]] = [
+                [] for _ in range(torch.distributed.get_world_size())
+            ]
+
+            torch.distributed.all_gather_object(saved_ops_names_all_ranks, objects[0])
             name_to_node = get_name_to_node(joint_graph)
-            saved_values = [name_to_node[n] for n in saved_values_names]
+            saved_sizes: list[int] = []
+            for saved_ops_names in saved_ops_names_all_ranks:
+                saved_nodes = [name_to_node[op_name] for op_name in saved_ops_names]
+                saved_size = 0
+                for node in saved_nodes:
+                    saved_size += _size_of(node)
+                saved_sizes.append(saved_size)
+
+            saved_sizes_tensor = torch.tensor(
+                saved_sizes, device=torch.cuda.current_device()
```

Comment:
Instead of hardcoding the cuda device here, can we try to use the same device that our collectives are going to use?

Maybe @wconstab knows of a cleaner way, but it looks like `all_gather_object` figures out what device to use with [this](https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L3153C5-L3153C52)
```
current_device = _get_object_coll_device(_group_or_default_group())
```